### PR TITLE
add caveexpress recipe

### PR DESCRIPTION
**Pull request by Ra-V2**
The recipe is currently broken, it's stop on 74% when building (on both x86 and x86_64). It's says something about SDL_RWclose. Any thoughts, what I forgot?

### CODE REVIEW

**Comment by Begasus** [comment]
Will check out the build today and report back to you

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+		caveexpress$secondaryArchSuffix = $portVersion
+		"
+REQUIRES="
+		haiku$secondaryArchSuffix
+		lib:liblua$secondaryArchSuffix
```
Comment (Begasus) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: Missing libatomic
```diff
+	lib:libatomic$secondaryArchSuffix
```

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		devel:libSDL2_net_2.0$secondaryArchSuffix
+		devel:libSDL2_image_2.0$secondaryArchSuffix
+		devel:libncurses$secondaryArchSuffix
+		devel:libsqlite3$secondaryArchSuffix
+		devel:libyajl$secondaryArchSuffix
+		"
+BUILD_PREREQUIRES="
+		cmd:cmake
+		cmd:gcc$secondaryArchSuffix
+		cmd:make
```
Comment (Begasus) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: Missing pkg-config:
```diff
+	cmd:pkg_config$secondaryArchSuffix
```

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+		devel:liblua$secondaryArchSuffix
+		devel:libBox2D$secondaryArchSuffix
+		devel:libSDL2_2.0$secondaryArchSuffix
+		devel:libSDL2_mixer_2.0$secondaryArchSuffix
+		devel:libSDL2_net_2.0$secondaryArchSuffix
+		devel:libSDL2_image_2.0$secondaryArchSuffix
+		devel:libncurses$secondaryArchSuffix
+		devel:libsqlite3$secondaryArchSuffix
+		devel:libyajl$secondaryArchSuffix
+		"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply games-arcade/caveexpress/caveexpress-2.4.recipe]: Sort alphabetically

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		lib:liblua$secondaryArchSuffix
+		lib:libBox2D$secondaryArchSuffix
+		lib:libSDL2_2.0$secondaryArchSuffix
+		lib:libSDL2_mixer_2.0$secondaryArchSuffix
+		lib:libSDL2_net_2.0$secondaryArchSuffix
+		lib:libSDL2_image_2.0$secondaryArchSuffix
+		lib:libncurses$secondaryArchSuffix
+		lib:libsqlite3$secondaryArchSuffix
+		lib:libyajl$secondaryArchSuffix
+		"
```
[the remark this answers is not in the document]
Comment (Begasus) [reply games-arcade/caveexpress/caveexpress-2.4.recipe]: Sort alphabetically

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+BUILD_REQUIRES="
+		haiku${secondaryArchSuffix}_devel
+		glm${secondaryArchSuffix}_devel
+		devel:liblua$secondaryArchSuffix
```
Comment (Begasus) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: Check for lua5.2:
```diff
-	devel:liblua$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix >= 5.2
```

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+CHECKSUM_SHA256="c1bf0707a4a2bcf2358b2791d97674a790d9fdce3ba29c695b9fa97030d74d3f"
+
+PATCHES="caveexpress-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	caveexpress$secondaryArchSuffix = $portVersion
+		"
```
Comment (Begasus) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: Only 1 tab :)

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	cmd:cmake
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build && cd build
+	cmake ..
```
Comment (Begasus) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: Add -L at the end of the line to show available options

**Comment by Ra-V2** [comment]
I finally built it on x86 but when I run it i got black screen and nothing shows.
screenshot1 (2) (1)

**Comment by Begasus** [comment]
You need to set a prefix (hint $appsDir), and maybe see if you can disable fullscreen (didn't go there yet) :) (make install should take care if installing the needed files, it needs more then just the binary)
caveexpress

**Comment by Ra-V2** [comment]
I try to build this with "make install PREFIX=$appsDir/caveexpress" and it gives me error "/boot/system/./cavepacker-icon.png": read-only file system. What I'm doing wrong?

**Comment by Begasus** [comment]
> I try to build this with "make install PREFIX=$appsDir/caveexpress" and it gives me error "/boot/system/./cavepacker-icon.png": read-only file system. What I'm doing wrong?

Mentione using cmake .. -L before, check for the option CMAKE_INSTALL_PREFIX

**Comment by Ra-V2** [comment]
Now it's creating folder with all files in $appsDir.

**Comment by Begasus** [comment]
> Now it's creating folder with all files in $appsDir.

That was the idea, there are 2 main applications, so I would suggest using $appsDir instead of $appsDir/caveexpress

**Comment by Ra-V2** [comment]
Should I create deskbar symlink for second application?

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+from your cave-dwelling clients and drop them off at the collection point. \
+But beware! Mighty mastodons, terrifying pterodactyls and others \
+would rather see you extinct."
+HOMEPAGE="https://caveproductions.org/"
+COPYRIGHT="2014-2016 Martin Gerhardy"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="https://github.com/mgerhardy/caveexpress/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="c1bf0707a4a2bcf2358b2791d97674a790d9fdce3ba29c695b9fa97030d74d3f"
+
```
Comment (korli) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: No newline here.

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+PATCHES="caveexpress-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	caveexpress$secondaryArchSuffix = $portVersion
+	"
```
Comment (korli) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: PROVIDES should provide something else.

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libBox2D$secondaryArchSuffix
+	devel:liblua$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
+	devel:libSDL2_image_2.0$secondaryArchSuffix
+	devel:libSDL2_mixer_2.0$secondaryArchSuffix
+	devel:libSDL2_net_2.0$secondaryArchSuffix
+	devel:libncurses$secondaryArchSuffix
```
Comment (korli) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: should come before devel:libSDL2_2.0

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+	cmd:luac$secondaryArchSuffix
+	cmd:make
+	cmd:pkg_config$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build && cd build
+	cmake .. -DCMAKE_INSTALL_PREFIX=$appsDir
+	make install
```
Comment (korli) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: make $jobArgs

File: games-arcade/caveexpress/caveexpress-2.4.recipe [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+INSTALL()
+{
```
Comment (korli) [inline games-arcade/caveexpress/caveexpress-2.4.recipe]: make install

Comment (Ra-V2) [reply games-arcade/caveexpress/caveexpress-2.4.recipe]: When I'm trying to build it says "no rule to make target 'install'"

Comment (Begasus) [reply games-arcade/caveexpress/caveexpress-2.4.recipe]: You need to change directory:
```diff
+	cd build
+	make install
```

Comment (Ra-V2) [reply games-arcade/caveexpress/caveexpress-2.4.recipe]: Oh, thanks I miss it.

**Comment by Ra-V2** [comment]
> Had make $jobArgs here before, removed it because it was taking up too much RAM so removed it back again, if it works ok for you, you can let it in.

I think it works fine for me. Building goes smoothly.

File: games-arcade/caveexpress/patches/caveexpress-2.4.patchset [hunk 13]
```diff
@@ -0,0 +1,228 @@
...
+ if (UNITTESTS)
+ 	add_subdirectory(src/tests)
+-endif()
++endif()
+\ No newline at end of file
+diff --git a/cmake/toolchains/haiku-toolchain.cmake b/cmake/toolchains/haiku-toolchain.cmake
+new file mode 100644
+index 0000000..7045206
+--- /dev/null
++++ b/cmake/toolchains/haiku-toolchain.cmake
```
Comment (Begasus) [inline games-arcade/caveexpress/patches/caveexpress-2.4.patchset]: Looks like this is a copy/paste from a merged PR by @extrowerk ?
https://github.com/mgerhardy/caveexpress/blob/a96cccb9ae5442514471e4901351b679470fcae1/cmake/toolchains/haiku-toolchain.cmake

**Comment by Begasus** [comment]
@Ra-V2 maybe checkout a latest commit (srcGitRev), I think this would resolve the major part of the patchset?

File: games-arcade/caveexpress/patches/caveexpress-2.4.patchset (hunk 13 above)
Comment (Ra-V2) [reply games-arcade/caveexpress/patches/caveexpress-2.4.patchset]: Exactly, this is it. I expected these files will download without srcGitRev but this not happend so I created patchset with these files.

**Comment by Ra-V2** [comment]
@Begasus I had to accidentally deleted this branch. Any thoughts how to push commit here now? Should I create new PR?

**Comment by Ra-V2** [comment]
I created a new one:
https://github.com/haikuports/haikuports/pull/4520

**Pull request closed without merging**